### ISSUE TITLE
unexpected NotImplementedException: ConvertToMySqlDbType for Enum is not implemented

### ISSUE THREAD

**Issue by MaceWindu** [mysql-net/MySqlConnector#850]
@bgrainger,

We had strange exception from our CI Linux run with v0.69.6 recently. Never seen it before, so it could be some tricky bug in provider. Decided to report it.
There were 16 failing tests, all on same schema read query

```sql
SELECT SPECIFIC_SCHEMA, SPECIFIC_NAME, PARAMETER_MODE, ORDINAL_POSITION, PARAMETER_NAME, NUMERIC_PRECISION, NUMERIC_SCALE, DATA_TYPE, CHARACTER_MAXIMUM_LENGTH, DTD_IDENTIFIER FROM INFORMATION_SCHEMA.parameters WHERE SPECIFIC_SCHEMA = database()
```

```
DataConnection: Error
Exception: MySql.Data.MySqlClient.MySqlException
Message  : Failed to read the result set.
   at MySql.Data.MySqlClient.MySqlDataReader.ActivateResultSet() in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlDataReader.cs:line 116
   at MySql.Data.MySqlClient.MySqlDataReader.CreateAsync(CommandListPosition commandListPosition, ICommandPayloadCreator payloadCreator, IDictionary`2 cachedProcedures, IMySqlCommand command, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlDataReader.cs:line 391
   at MySqlConnector.Core.CommandExecutor.ExecuteReaderAsync(IReadOnlyList`1 commands, ICommandPayloadCreator payloadCreator, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/Core/CommandExecutor.cs:line 62
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteDbDataReader(CommandBehavior behavior) in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlCommand.cs:line 210
   at System.Data.Common.DbCommand.ExecuteReader(CommandBehavior behavior)
   at LinqToDB.Data.DbCommandProcessor.DbCommandProcessorExtensions.ExecuteReaderExt(IDbCommand cmd, CommandBehavior commandBehavior) in d:\a\1\s\Source\LinqToDB\Data\DbCommandProcessor\DbCommandProcessorExtensions.cs:line 42
   at LinqToDB.Data.DataConnection.ExecuteReader(IDbCommand command, CommandBehavior commandBehavior) in d:\a\1\s\Source\LinqToDB\Data\DataConnection.cs:line 1387
   at LinqToDB.Data.DataConnection.ExecuteReader(CommandBehavior commandBehavior) in d:\a\1\s\Source\LinqToDB\Data\DataConnection.cs:line 1419

Exception: System.NotImplementedException
Message  : ConvertToMySqlDbType for Enum is not implemented
   at MySqlConnector.Core.TypeMapper.ConvertToMySqlDbType(ColumnDefinitionPayload columnDefinition, Boolean treatTinyAsBoolean, MySqlGuidFormat guidFormat) in /_/src/MySqlConnector/Core/TypeMapper.cs:line 276
   at MySqlConnector.Core.ResultSet.ReadResultSetHeaderAsync(IOBehavior ioBehavior) in /_/src/MySqlConnector/Core/ResultSet.cs:line 148
```

link to one of failed tests : https://dev.azure.com/linq2db/linq2db/_build/results?buildId=2690&view=ms.vss-test-web.build-test-results-tab&runId=50096&resultId=101991&paneView=attachments

Full nunit test log
[_fv-az774_2020-07-15_13_47_49.zip](https://github.com/mysql-net/MySqlConnector/files/4931928/_fv-az774_2020-07-15_13_47_49.zip)

**Comment by MaceWindu** [comment]
Actually it could be caused by new version of mysql. We use [mysql:latest](https://hub.docker.com/layers/mysql/library/mysql/latest/images/sha256-6b6cd9550cb51f0d188bc1d3244464feaaba222dad53c7686d721ed88849de37?context=explore) image, which was updated just recently to 8.0.21.

Probably now parameters table use enum (e.g. for PARAMETER_MODE column), but cannot see it in release notes https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-21.html

**Comment by bgrainger** [comment]
I can confirm that `INFORMATION_SCHEMA.parameters.PARAMETER_MODE` has changed from `CHAR(5)` to `ENUM` between MySQL 8.0.20 and MySQL 8.0.21, and that does appear to be triggering the bug.

**Comment by bgrainger** [comment]
The integration tests already test `ENUM` columns but apparently the metadata for the `PARAMETER_MODE` column is different.

https://github.com/mysql-net/MySqlConnector/blob/4cd45f6e1a682b579a05b91aed4e80151b6bf33a/tests/SideBySide/DataTypesFixture.cs#L45-L55

**Comment by bgrainger** [comment]
The `parameters.PARAMETER_MODE` column metadata:
![image](https://user-images.githubusercontent.com/188129/87704217-eb775980-c750-11ea-9797-6243a26ea730.png)

The `datatypes_enums.size` column metadata:
![image](https://user-images.githubusercontent.com/188129/87704138-c71b7d00-c750-11ea-8e28-fe73d2511904.png)

I don't yet know why these are different.

**Comment by bgrainger** [comment]
This is fixed in [0.69.7](https://github.com/mysql-net/MySqlConnector/releases/tag/0.69.7); there's no fix in the 1.x branch yet.

Thanks for the concise and accurate bug report!

**Comment by bgrainger** [comment]
Fix for 1.x is in 1.0.1-beta.0.8 in [GitHub packages](https://github.com/mysql-net/MySqlConnector/packages/39735).

**Comment by bgrainger** [comment]
Fixed in [1.0.1](https://github.com/mysql-net/MySqlConnector/releases/tag/1.0.1).
